module/filter/doubleFilter.py

```diff
@@ -10,33 +10,45 @@
                      by Bastian Schroll
 
 @file:        doubleFilter.py
-@date:        15.01.2018
-@author:      Bastian Schroll
-@description: Class to implement a filter for double alarms
-@todo test, refactor and document / check_msg is not implemented yet
+@date:        09.07.2020
+@author:      Bastian Schroll, b-watch
+@description: Filter module for double packages
 """
 import logging
+from module.moduleBase import ModuleBase
+
+# ###################### #
+# Custom plugin includes #
 import time
+# ######################
 
 logging.debug("- %s loaded", __name__)
 
 
-class DoubleFilter:
-    """!Double Filter Class"""
-
+class BoswatchModule(ModuleBase):
+    """!Description of the Module"""
     def __init__(self, config):
-        """!init"""
-        self._config = config
+        """!Do not change anything here!"""
+        super().__init__(__name__, config)  # you can access the config class on 'self.config'
         self._filterLists = {}
+        logging.debug("Configured ignoreTime: %d", self.config.get("ignoreTime", default=10))
+        logging.debug("Configured maxEntry: %d", self.config.get("maxEntry", default=10))
 
-    def filter(self, bwPacket):
+    def onLoad(self):
+        """!Called by import of the plugin
+        Remove if not implemented"""
+        pass
 
+    def doWork(self, bwPacket):
+        """!start an run of the module.
+
+        @param bwPacket: A BOSWatch packet instance"""
         if bwPacket.get("mode") == "fms":
-            scanWord = "fms"
+            filterFields = ["fms"]
         elif bwPacket.get("mode") == "pocsag":
-            scanWord = "ric"
+            filterFields = self.config.get("pocsagFields", default=["ric"])
         elif bwPacket.get("mode") == "zvei":
-            scanWord = "zvei"
+            filterFields = ["zvei"]
```

Comment:
Müsste hier nicht statt "zvei" "tone" stehen, sodass "filterFields" der 5-Ton-Folge entspricht?

Reply:
Da gab es mal ein Issue zu, kann mich wage dran erinnern.
Ich habe die Doku auch mal behoben und 'zvei' zu 'tone' umgeschrieben, also dann vermutlich eher so:

```suggestion
            filterFields = ["tone"]
```

Danke für den Einwand, ich glaube das wäre sowohl @Schrolli91 und vor allem nicht mir aufgefallen :)

Reply:
Das stimmt, Danke für den Hinweis. Ist angepasst.

Reply:
@janspeller habs grad nochmal in der Doku geprüft - die Anpassungen sind auf jeden Fall schon online
https://docs.boswatch.de/develop/packet.html#speziell-fur-zvei